test/unittests/skills/test_audioservice.py

```diff
@@ -0,0 +1,162 @@
+import unittest
+from unittest import mock
+
+from mycroft.messagebus import Message
+from mycroft.skills.audioservice import AudioService
+
+
+class TestAudioService(unittest.TestCase):
+    def test_lifecycle(self):
+        bus = mock.Mock(name='bus')
+        audioservice = AudioService(bus)
+        self.assertEqual(audioservice.bus, bus)
+
+    def test_play(self):
+        bus = mock.Mock(name='bus')
+        audioservice = AudioService(bus)
+
+        # Test proper uri
+        audioservice.play('file:///hello_nasty.mp3')
+        message = bus.emit.call_args_list[-1][0][0]
+        self.assertEqual(message.msg_type, 'mycroft.audio.service.play')
+        self.assertEqual(message.data['tracks'], ['file:///hello_nasty.mp3'])
+        self.assertEqual(message.data['repeat'], False)
+
+        # Test path
+        audioservice.play('/hello_nasty.mp3')
+        message = bus.emit.call_args_list[-1][0][0]
+        self.assertEqual(message.msg_type, 'mycroft.audio.service.play')
+        self.assertEqual(message.data['tracks'], ['file:///hello_nasty.mp3'])
+        self.assertEqual(message.data['repeat'], False)
+
+        # Test tuple
+        audioservice.play(('/hello_nasty.mp3', 'audio/mp3'))
+        message = bus.emit.call_args_list[-1][0][0]
+        self.assertEqual(message.msg_type, 'mycroft.audio.service.play')
+        self.assertEqual(message.data['tracks'],
+                         [('file:///hello_nasty.mp3', 'audio/mp3')])
+        self.assertEqual(message.data['repeat'], False)
+
+        # test invalid
+        with self.assertRaises(ValueError):
+            audioservice.play(12)
+
+        # Test extra arguments
+        audioservice.play('/hello_nasty.mp3', 'on vlc', True)
+        message = bus.emit.call_args_list[-1][0][0]
+        self.assertEqual(message.msg_type, 'mycroft.audio.service.play')
+        self.assertEqual(message.data['tracks'], ['file:///hello_nasty.mp3'])
+        self.assertEqual(message.data['repeat'], True)
+        self.assertEqual(message.data['utterance'], 'on vlc')
+
+    def test_queue(self):
+        bus = mock.Mock(name='bus')
+        audioservice = AudioService(bus)
+
+        # Test proper uri
+        audioservice.queue('file:///hello_nasty.mp3')
```

Comment:
lots of duplicate code here. consider a function where you pass an argument for the queue and one for the actual value in the second assert statement